src/components/checkbox/checkbox.js

```diff
@@ -92,9 +92,22 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       post: postLink
     };
 
-    function postLink(scope, element, attr, ngModelCtrl) {
+    function postLink(scope, element, attr, ctrls) {
       var isIndeterminate;
-      ngModelCtrl = ngModelCtrl || $mdUtil.fakeNgModel();
+      var containerCtrl = ctrls[0];
+      var ngModelCtrl = ctrls[1] || $mdUtil.fakeNgModel();
+      var formCtrl = ctrls[2];
+
+      if (containerCtrl) {
+        var isErrorGetter = containerCtrl.isErrorGetter || function() {
+          return ngModelCtrl.$invalid && (ngModelCtrl.$touched || (formCtrl && formCtrl.$submitted));
+        };
+
+        containerCtrl.input = element;
+
+        scope.$watch(isErrorGetter, containerCtrl.setInvalid);
```

Comment:
I used the same approach as [input.js](https://github.com/angular/material/blob/fc7e9b3fc87713c6fde3b82c0df358650ec9aafc/src/components/input/input.js#L358), [datepickerDirective.js](https://github.com/angular/material/blob/a73ab43c437ec7468d51da54f159b9f2845ec07b/src/components/datepicker/js/datepickerDirective.js#L163-L165) and [select.js](https://github.com/angular/material/blob/a73ab43c437ec7468d51da54f159b9f2845ec07b/src/components/select/select.js#L302). I did not look for another solution.